### ISSUE TITLE
Make Image Containers links on Detail Page

### ISSUE THREAD

**Issue by JamWils** [weaveworks/weave-gitops#2267]
We would want all docker images displayed to be links, but as discovered in #2326, different image repositories need to be linked to in different ways.

**Acceptance criteria**
Implement support for (these examples all come from #2326, look there for more of a description):
 * Dockerhub - all formats. E.g.
   * docker.io/weaveworks/eksctl -> https://hub.docker.com/r/weaveworks/eksctl
   * weaveworks/eksctl -> https://hub.docker.com/r/weaveworks/eksctl
   * docker.io/nginx -> https://hub.docker.com/r/_/nginx
   * nginx -> https://hub.docker.com/r/_/nginx
   * library/nginx -> https://hub.docker.com/r/_/nginx
   * docker.io/library/nginx -> https://hub.docker.com/r/_/nginx
  * Quay.io. E.g.:
    * quay.io/jitesoft/nginx -> https://quay.io/repository/jitesoft/nginx
   * GHCR. E.g.:
     * ghcr.io/weaveworks/charts/weave-gitops can just be used.
   * GCR. E.g.:
     * gcr.io/cloud-builders/gcloud can just be used.
 
This will be reused in more places, such as for OCI helmcharts and the flux runtime page, so please make sure it can be.

**Out of scope**
We do not want to link to the exact tag - just the image. Everything after the colon is always stripped out from the link

We will not right now link to image repositories we can't recognise - if it's not one of these cases, just print it as text.

We will not right now make this user-configurable.

**Comment by ozamosi** [comment]
@JamWils the details table has images for pods etc - is this about making them links, or do they need to be added somewhere else?

**Comment by JamWils** [comment]
Yeah, @ozamosi this was an intended placeholder to make them links.  I never filled it out.  I believe @chanwit did a spike.  I'll review it.

**Comment by ozamosi** [comment]
:+1: then I understand what this wants us to do.

I've summed up and filled in a suggestion from #2326, and I think that's sufficient to get going.

I mostly decided on "Don't link to unknown registries" on a coin flip - on one hand, the data we have suggest that "just" linking will usually not work, on the other hand we've looked at big and old ones so there's definitely sampling bias. It's a default, it can be changed, feel free to change/clarify.
